haskell/test-common/Lorentz/Contracts/Test/Common.hs

```diff
@@ -235,24 +238,71 @@ mkInitialStorage OriginationParams{..} metadataRegistryAddress mContractMetadata
 mgmContractPaused :: ExecutorError -> IntegrationalScenario
 mgmContractPaused = lExpectFailWith (== [mt|CONTRACT_PAUSED|])
 
-originateMetadataRegistry :: IntegrationalScenarioM Address
-originateMetadataRegistry =
-  tOriginate
-    registryContract
-    "Metadata Registry contract"
-    (toVal defaultMetadataRegistryStorage)
-    (toMutez 0)
-
-nettestOriginateMetadataRegistry :: MonadNettest caps base m => m Address
-nettestOriginateMetadataRegistry =
-  originateUntypedSimple
-    "nettest.MetadataRegistry"
-    (untypeValue (toVal defaultMetadataRegistryStorage))
-    (convertContract registryContract)
-
 nettestOriginateContractMetadataContract :: (ToJSON metadata) => MonadNettest caps base m => metadata -> m Address
 nettestOriginateContractMetadataContract mdata =
   originateUntypedSimple
     "nettest.ContractMetadata"
     (untypeValue (toVal $ mkContractMetadataRegistryStorage $  metadataMap (CurrentContract mdata False)))
     (convertContract contractMetadataContract)
+
+checkView
+  :: forall viewParam viewVal parameter
+   . (IsoValue viewParam, IsoValue viewVal, Eq viewVal, Show viewVal)
+  => TAddress parameter -> Text -> viewParam -> viewVal -> IntegrationalScenario
+checkView addr viewName viewParam expectedViewVal =
```

Comment:
Correct me if I'm wrong, but isn't most of this functions's check generally about `TZIP-16` and not this specific contract?

Do you think we could move it to `morley-metadata` to reduce duplication (if not now in the future)?

Reply:
Have raised a PR there https://gitlab.com/morley-framework/morley-metadata/-/merge_requests/11 .